docs/guides/4-custom-rulesets.md

```diff
@@ -343,7 +402,51 @@ aliases:
   Paths: "$.paths[*]~"
 ```
 
-Rulesets can then reference aliases in the [given](#given) keyword, either in full: `"given": "#Paths"`, or use it as a prefix for further JSON Path syntax, like dot notation: `"given": "#InfoContact.name"`.
+If you deal with a variety of different spec, you may find the above approach insufficient, particularly when the shape of the document is notably different.
+In such a case, you may want to consider using scoped aliases.
+
+```yaml
+aliases:
+  SharedParameterObject:
+    description: an optional property describing the purpose of the alias
+    targets:
+      - formats:
+          - oas2
+        given: $.parameters[*]
+      - formats:
+          - oas3
+        given: $.components.parameters[*]
+```
+
+Now, if you referenced `SharedParameterObject` alias, the chosen path would be determined based on the document you use.
+For instance, if a given document matched OpenAPI 2.x, `$.parameters[*]` would be used as the JSONPath expression.
+
+Having a closer look on the example above, one may notice that it'd be still somewhat complicated to target _all_ Parameter Objects
+that a specific OpenAPI document may contain.
+To make it more feasible and avoid overly complex JSONPath expressions, `given` can be an array.
+
+```yaml
+aliases:
+  PathItemObject: $.paths[*]
+  OperationObject: "#PathItem[get,put,post,delete,options,head,patch,trace]"
+  ParameterObject:
+    description: an optional property describing the purpose of the alias
+    targets:
+      - formats:
+          - oas2
+        given:
+          - "#PathItemObject.parameters[*]"
+          - "#OperationObject.parameters[*]"
+          - $.parameters[*]
+      - formats:
+          - oas3
+        given:
+          - "#PathItemObject.parameters[*]"
+          - "#OperationObject.parameters[*]"
+          - $.components.parameters[*]
+```
+
+Rulesets can then reference aliases in the [given](#given) keyword, either in full: `"given": "#Paths"`, or use it as a prefix for further JSONPath syntax, like dot notation: `"given": "#InfoContact.name"`.
```

Comment:
Very neat - perhaps adjust this to use an example from above? E.g. `"given": "#PathItemObject"` and `"given": "#ParameterObject.name"` or something? just easier to connect the dots